src/Tokenizer/Tokens.php

```diff
@@ -1088,17 +1088,19 @@ public function clearRange(int $indexStart, int $indexEnd): void
      */
     public function isMonolithicPhp(): bool
     {
-        $size = $this->count();
-
-        if (0 === $size) {
+        if (0 === $this->count()) {
             return false;
         }
 
-        if ($this->isTokenKindFound(T_INLINE_HTML)) {
+        if ($this->countTokenKind(T_INLINE_HTML) > 1) {
             return false;
         }
 
-        return 1 >= ($this->countTokenKind(T_OPEN_TAG) + $this->countTokenKind(T_OPEN_TAG_WITH_ECHO));
+        if (1 === $this->countTokenKind(T_INLINE_HTML)) {
+            return 1 === Preg::match('/^#!.*$/', $this[0]->getContent());
```

Comment:
```suggestion
            return 1 === Preg::match('/^#!.+$/', $this[0]->getContent());
```

to handle https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6233/files#r784369829